# VQVAE Configuration for CrowdSurfer

model:
  perception_encoder:
    input_type: 'laserscan'
    input_dim: 360
    output_dim: 512
    hidden_dims: [256, 512, 512]
    use_attention: true

  vqvae:
    perception_dim: 512
    goal_dim: 3  # (x, y, theta)
    encoder_hidden: [256, 128]
    decoder_hidden: [128, 256]
    latent_dim: 64
    num_embeddings: 512
    commitment_cost: 0.25
    horizon: 30  # 3 seconds at 10Hz
    output_dim: 2  # (x, y)

  pixelcnn:
    num_embeddings: 512
    num_layers: 6
    hidden_channels: 128
    kernel_size: 5
    dropout: 0.1
    condition_dim: 3

training:
  batch_size: 128
  learning_rate: 0.0003
  num_epochs: 200
  warmup_steps: 1000
  weight_decay: 0.0001

  optimizer: 'adam'
  scheduler: 'cosine'

  checkpoint_interval: 10
  validation_interval: 5

  device: 'cuda'
  mixed_precision: false

data:
  horizon: 3.0  # seconds
  dt: 0.1  # seconds
  min_trajectory_length: 20
  max_trajectory_length: 50

  augmentation:
    mirror: true
    rotate: true
    scale: false
